vocode/streaming/utils/worker.py

```diff
@@ -36,6 +35,43 @@ def terminate(self):
         return False
 
 
+class ThreadAsyncWorker(AsyncWorker):
```

Comment:
The way I was initially envisioning ThreadAsyncWorker was to have the async worker run in the front (so it looks like a regular worker)  and then still run the blocking calls in a thread each time a new sentence came in (vs a continuous task). I realize that the pattern set up in Azure probably makes it not necessary anymore as long as you're returning SynthesisResult with an async generator (vs putting items on a queue).

Why not follow the same pattern for blocking calls? What's the problem the approach you used in Azure?  The main benefit is that we can easily encapsulate the logic to run multiple blocking synthesizers concurrently without the synthesizer implementing that logic.

Reply:
it just means that we can implement _run_loop without async and it should be okay. makes it easier for folks to implement transcribers like Google, Azure
- a lot of the guides for transcription services from major cloud providers are not async

Reply:
Got it. I wonder how those transcribers work without using a async websocket connection. For synthesizers, I think it's easier, more efficient and as fast to follow the pattern established in the Azure.